SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorPackageAssembly.cpp

```diff
@@ -0,0 +1,426 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKEditorPackageAssembly.h"
+
+#include "CoreMinimal.h"
+
+#include "Async/Async.h"
+#include "Misc/App.h"
+#include "Misc/FileHelper.h"
+#include "UnrealEdMisc.h"
+#include "EditorStyle.h"
+#include "SpatialGDKSettings.h"
+#include "SpatialGDKEditorModule.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKServicesModule.h"
+
+#include "UATHelper/Public/IUATHelperModule.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKEditorPackageAssembly);
+
+#define LOCTEXT_NAMESPACE "SpatialGDKEditorPackageAssembly"
+
+
+
+FSpatialGDKPackageAssembly::FSpatialGDKPackageAssembly() : CurrentAssemblyTarget{ EPackageAssemblyTarget::NONE }
+{
+
+}
+
+static FString GetStagingDir()
+{
+	return FPaths::ConvertRelativePathToFull((FPaths::IsProjectFilePathSet() ? FPaths::GetPath(FPaths::GetProjectFilePath()) : FPaths::RootDir() / FApp::GetProjectName()) / TEXT("..") / TEXT("spatial") / TEXT("build") / TEXT("unreal"));
+}
+
+static void WriteStartWorkerScript()
```

Comment:
I think we're duplicating too much code from `Build.cs` here. We should either be calling `Build.cs` from the editor to handle the building and packaging for maintainability or we completely replace `Build.cs` and it's functionality, this would require commandline access and updating all our existing build pipelines.

Reply:
Can we approach this iteratively? e.g. start with this, add code to make it a commandlet, then phase out build.cs and update CI? or I can change this to use build.bat instead of using the IUATHelper, however the helper manages the notifications and cancellations by itself. What's the goal of going forward for Build.cs?

One of the arguments for removing Build.cs is that it uses xcopy which is a DOS/Windows utility, leveraging IFilePlatform makes for code that works cross platform without needing anything special from us.

Also we need this file to have code for discovering the Unreal Engine, which is again platform dependent because it uses the Registry.

I've been told there's some constraint that linux builds can't be made on OS X? So perhaps being cross platform isn't important?

Reply:
So I think the minimum would be to use the `Build.cs` to prevent code duplication. Our customers also rely on Build.cs so offering them a new tool which does the same thing and becomes a place they have to potentially modify in two locations is confusing. 
@jessicafalk Is best placed to talk about the OSX work. 
As for the `xcopy` we can change this to anything we want really, I think it's already been changed for Mac builds.

Reply:
Alright a bit of offline communication, I'll switch this to call the build.exe that build.cs creates.